packages/nimble-components/src/select/specs/option-groups-hld.md

```diff
@@ -0,0 +1,79 @@
+# Select and Combobox option groups
+
+## Problem Statement
+
+It is not uncommon to want to categorize the options that are present in the dropdown for a `Select`/`Combobox` component. By supporting categories, users are then able to provide options that have the same display value, but under separate categories, helping avoid confusion.
+
+## Links To Relevant Work Items and Reference Material
+
+-   [Interaction Design](./IxD.md#groups)
+-   [Sectioned Combobox (issue 791)](https://github.com/ni/nimble/issues/791)
+-   [Prototype branch Storybook](https://60e89457a987cf003efc0a5b-cvmxlqwglb.chromatic.com/?path=/docs/components-select--docs) ([code](https://github.com/ni/nimble/tree/groupable-select))
+
+## Implementation / Design
+
+The HTML native `select` supports option groups with the native `optgroup` element. Configuring the `select` to use the `optgroup` takes the following form:
+
+```html
+<select name="cars">
+    <optgroup label="Swedish Cars">
+        <option value="volvo">Volvo</option>
+        <option value="saab">Saab</option>
+    </optgroup>
+    <optgroup label="German Cars">
+        <option value="mercedes">Mercedes</option>
+        <option value="audi">Audi</option>
+    </optgroup>
+</select>
+```
+
+The Nimble `Select` (and `Combobox`) will follow the same approach, and provide a new custom element, `nimble-list-option-group`, that mirrors the `optgroup` behavior/API:
+
+```html
+<nimble-select name="cars">
+    <nimble-list-option-group label="Swedish Cars">
+        <nimble-list-option value="volvo">Volvo</option>
+        <nimble-list-option value="saab">Saab</option>
+    </nimble-list-option-group>
+    <nimble-list-option-group label="German Cars">
+        <nimble-list-option value="mercedes">Mercedes</option>
+        <nimble-list-option value="audi">Audi</option>
+    </nimble-list-option-group>
+</nimble-select>
+```
+
+### API
+
+The `ListOptionGroup` class API should be fairly lightweight:
+
+```ts
+public class ListOptionGroup {
+    @attr
+    public label?: string;
+}
+```
+
+_Question: Should we support slotting icons (other content?) into the group display?_
+
+### Filtering
+
+As mentioned in the IxD, when a user has filtering enabled, typing into the filter input will match both individual options (as normal), as well as _all_ options that are under a group that matches the filter text.
+
+_Question: Should matching against a group be configurable?_
```

Comment:
I agree – I challenged the other POs on the requirement to search groups, and they continue to be interested in that capability. However, it's not the common pattern, so we should be as consistent as possible so that our users understand the behavior and capability.